gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -300,6 +300,10 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
       }
     }
 
+    // execute post publish commit steps after preserving file attributes, because some post publish step,
+    // e.g. SetPermissionCommitStep needs to set permissions
+    executeCommitSequence(postPublish);
```

Comment:
We do move path from tmp to dst in this step, I believe it should happen before the set file attribute. We should make sure set file attribute preserve the right permission

Reply:
moving from tmp(writer.output.dir actually) to dst is publish step, right?